sw/cheri/boot/boot_loader.cc

```diff
@@ -128,13 +153,42 @@ extern "C" void rom_loader_entry(void *rwRoot)
 	sram.address()                  = SRAM_ADDRESS + 0x1000;
 	sram.bounds()                   = SRAM_BOUNDS - 0x1000;
 
+	CHERI::Capability<uint8_t> hyperram = root.cast<uint8_t>();
+	hyperram.address()                  = HYPERRAM_ADDRESS;
+	hyperram.bounds()                   = HYPERRAM_BOUNDS;
+
 	spi->init(false, false, true, 0);
 	uart->init(BAUD_RATE);
 
 	SpiFlash spi_flash(spi, gpio, FLASH_CSN_GPIO_BIT);
 	spi_flash.reset();
-	read_elf(spi_flash, uart, sram);
+	uint32_t entrypoint = read_elf(spi_flash, uart, sram, hyperram);
 
 	write_str(uart, prefix);
 	write_str(uart, "Booting into program, hopefully.\r\n");
+	return entrypoint;
+}
+
+#define READ_CSR(name) ({ \
```

Comment:
Please could you add a comment pointing towards https://gcc.gnu.org/onlinedocs/gcc/Statement-Exprs.html